ImageViewer/Source/GalleryPagingDataSource.swift

```diff
@@ -16,7 +16,7 @@ final class GalleryPagingDataSource: NSObject, UIPageViewControllerDataSource {
 
     fileprivate let configuration: GalleryConfiguration
     fileprivate var pagingMode = GalleryPagingMode.standard
-    fileprivate let itemCount: Int
+    fileprivate var itemCount: Int { return itemsDataSource!.itemCount() }
```

Comment:
When I started with Swift, I took a very strict approach with optionals, essentially never force unwrapping.  I've since relaxed that view.  In cases where optionals are guaranteed to be non-nil or would be a programmer error if they were, force unwrapping and the resultant crash are the way to go, as it allows finding the problem early.  I found `guard let` with `fatalError()` to be unnecessary extra code.  

In this case, attempting to use this class without an itemsDataSource defined would be a programmer error.

Another point is that propagating optionals can result in unnecessary complexity as the optional must be dealt with at multiple levels, so I avoid returning optionals unless it really makes sense for the interface.